pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -518,6 +518,11 @@ func convertToPrincipal(subject *rbacproto.Subject) *policyproto.Principal {
 
 		for _, k := range keys {
 			v := subject.Properties[k]
+			if k == attrSrcPrincipal && subject.User != "" {
+				rbacLog.Errorf("ignored %s, duplicate with previous user value %s",
+					attrSrcPrincipal, subject.User)
+				continue
+			}
```

Comment:
Without this logic, it basically means the `source.principal` will override the `user` value when `user` is not empty and `*`.

Could you clarify about what extra flexibility do you mean here?

Reply:
I was thinking of having multiple entries of `source.principal` to specify the use cases such as "allow all users with prefix `prod` except `abc`". But given that we haven't added support for "excludedValues", and this can be done by specifically creating two "source.principal" entries, I think your current change makes sense.